dhcp-relay/dhcp_kern_xdp.c

```diff
@@ -438,57 +631,50 @@ int xdp_dhcp_relay(struct xdp_md *ctx) {
 
 			if (write_dhcp_option_255(ctx, option_offset)) {
 				bpf_printk("Could not write DHCP option 255 at offset %i", option_offset);
-				return XDP_ABORTED;
+				//return XDP_ABORTED;
+				break;
 			}
 
 			bpf_printk("Wrote DHCP option 255 at offset %i, returning XDP_PASS", option_offset);
 
 			break;
 		}
+
 		pos++;
 
+		/* Verifier check */
+		if (pos + 1 > data_end) {
+			break;
+		}
+
 		option_length = *pos;
 		option_offset += option_length + 2;
 
+		/* Verifier check */
 		if (pos + 1 > data_end) {
 			break;
 		}
 		pos++;
 
+		/* Verifier check */
```

Comment:
For instance, the check below could be moved up, and changed to `if (pos + option_length + 1 > data_end)`